### PR TITLE
Add functionality for tool to be run as module from command line.

### PR DESCRIPTION
**Pull request by isaac-tfn**
Describe Changes
---

Task Checklist
----
- [ ] Have unittests been added (testing individual functions and their edge cases)
- [ ] Have integration tests been added (if applicable, to test modules of functionality)
- [x] Updated RELEASE.md to reflect changes in PR
- [x] Have new dependencies been added?
  - [x] Have they been added to either `requirements.txt` or `requirements_dev.txt`.
  - [x] Have they been added to `setup.cfg`

### CODE REVIEW

File: src/caf/space/inputs.py [hunk 1]
```diff
@@ -157,6 +161,59 @@ def _valid_data_col(cls, v, values):
...
+        """Parse command line argument."""
+        parser = argparse.ArgumentParser(
+            description=__doc__, formatter_class=argparse.ArgumentDefaultsHelpFormatter
+        )
+        parser.add_argument(
+            "--mode",
+            type=str,
+            help="Mode to run translation in; spatial, weighted or GUI.",
+            default="GUI",
+            required=False,
```
Comment (wsp-mbuckley) [inline src/caf/space/inputs.py:183]: `add_argument` has a choices parameters which will make sure the input is one of a limited set of options e.g. `choices=MODES` will check the given value is in `MODES`.

File: src/caf/space/inputs.py [hunk 2]
```diff
@@ -157,6 +161,59 @@ def _valid_data_col(cls, v, values):
...
+    out_path: Path
+
+    @classmethod
+    def parse(cls) -> SpaceArguments:
+        """Parse command line argument."""
+        parser = argparse.ArgumentParser(
+            description=__doc__, formatter_class=argparse.ArgumentDefaultsHelpFormatter
+        )
+        parser.add_argument(
+            "--mode",
```
Comment (wsp-mbuckley) [inline src/caf/space/inputs.py:179]: Adding first parameter `"-m"` (before `"--mode"`) will allow the user to specify using either `-m` or `--mode` to make arguments shorter

File: src/caf/space/inputs.py [hunk 3]
```diff
@@ -157,6 +161,59 @@ def _valid_data_col(cls, v, values):
...
+    mode: str
+    out_path: Path
+
+    @classmethod
+    def parse(cls) -> SpaceArguments:
+        """Parse command line argument."""
+        parser = argparse.ArgumentParser(
+            description=__doc__, formatter_class=argparse.ArgumentDefaultsHelpFormatter
+        )
+        parser.add_argument(
```
Comment (wsp-mbuckley) [inline src/caf/space/inputs.py:178]: I don't think this is necessary here but for future reference you can use `add_subparser` to have separate sub-commands which have their own set of arguments (and some shared arguments), see [argparse sub-commands](https://docs.python.org/3/library/argparse.html#sub-commands).

For example you could have sub-command "GUI" with no arguments and sub-commands sub-commands "spatial" and "weighted" which have mandatory config path and output folder arguments.

File: src/caf/space/inputs.py [hunk 4]
```diff
@@ -157,6 +161,59 @@ def _valid_data_col(cls, v, values):
         return v
 
 
+@dataclasses.dataclass
+class SpaceArguments:
+    """Command Line arguments for running space."""
+
+    config_path: Path
+    mode: str
```
Comment (wsp-mbuckley) [inline src/caf/space/inputs.py:169]: A custom Enum class would make this variable more obvious what it's possible values are.

File: src/caf/space/inputs.py [hunk 5]
```diff
@@ -157,6 +161,59 @@ def _valid_data_col(cls, v, values):
...
+        """Raise error for invalid input."""
+        if self.config_path:
+            if not self.config_path.is_file():
+                raise FileNotFoundError(f"config file doesn't exist: {self.config_path}")
+
+        if self.out_path:
+            if not self.out_path.is_dir():
+                raise FileNotFoundError(f"{self.out_path} does not exist.")
+
+        if self.mode not in MODES:
```
Comment (wsp-mbuckley) [inline src/caf/space/inputs.py:213]: Could use `choices` parameter in `add_argument` method instead of this

**Review by wsp-mbuckley — APPROVED** [review]
This all looks good, I've added a few comments where there's some features of argparse which might be worth using next time but I don't think those changes need to be done right now as the arguments are very simple and work well.
